### PR TITLE
feat(library): Add support for Caddy

### PR DESCRIPTION
**Pull request by razvand**
Introduce Caddy bincompat run. Build Caddy as static PIE using a Dockerfile, inspired by Caddy's own Docker setup. Use `xcaddy` to build. The build uses binary compatibility mode (i.e. the `base` image).

Add:

* `Kraftfile`: build / run rules, including pulling the `base` image
* `Dockerfile`: base filesystem, with binary and libraries
* `README.md`: document how to use
* `Caddyfile`: Caddy configuration file

### CODE REVIEW

File: library/caddy/2.7/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,32 @@
+FROM --platform=linux/x86_64 golang:1.21.4-bookworm AS build
+
+# Inspired from: https://github.com/caddyserver/caddy-docker/blob/master/2.7/builder/Dockerfile
+
+RUN set -eux; apt-get update ; \
+	apt-get -yqq install ca-certificates ; \
+	apt-get -yqq install git ; \
+	apt-get -yqq install libcap2-bin
```
Comment (nderjung) [inline library/caddy/2.7/Dockerfile:8]:
```Dockerfile
RUN set -xe; 
    apt-get update; \
    apt-get install -y --no-install-recommends \
      ca-certificates \
      git \
      libcap2-bin \
    ;
```

File: library/caddy/2.7/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+ENV XCADDY_SETCAP 1
+
+RUN set -eux; \
+	wget -O /tmp/xcaddy.tar.gz "https://github.com/caddyserver/xcaddy/releases/download/v0.3.5/xcaddy_0.3.5_linux_amd64.tar.gz"; \
+	tar x -z -f /tmp/xcaddy.tar.gz -C /usr/bin xcaddy; \
+	rm -f /tmp/xcaddy.tar.gz; \
+	chmod +x /usr/bin/xcaddy;
+
+env CGO_ENABLED=1
+env XCADDY_GO_BUILD_FLAGS "-buildmode=pie -ldflags '-linkmode external -extldflags -static-pie'"
```
Comment (nderjung) [inline library/caddy/2.7/Dockerfile:25]: Uppercase `ENV`

File: library/caddy/2.7/Kraftfile [hunk 3]
```diff
@@ -0,0 +1,7 @@
+spec: v0.6
+
+runtime: index.unikraft.io/official/base:latest
```
Comment (nderjung) [inline library/caddy/2.7/Kraftfile:3]:
```yaml
runtime: unikraft.org/base:latest
```

File: library/caddy/2.7/README.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
+# Caddy 2.7
+
+This directory contains the definition for the `unikraft.org/caddy:2.7` image.
+
+To run this image, [install Unikraft's companion command-line toolchain `kraft`](https://unikraft.org/docs/cli) and then you can run:
+
+```
+kraft run unikraft.org/caddy:2.7 -p 2015:2025
```
Comment (nderjung) [inline library/caddy/2.7/README.md:8]: `-p 2015:2015`

File: library/caddy/2.7/Dockerfile (hunk 1 above)
Comment (nderjung) [reply library/caddy/2.7/Dockerfile]: Please use this exactly, also with indentation

**Review by StefanJum — CHANGES_REQUESTED** [review]
The README needs to be updated with the build and run commands like the other prs. Everything else works.

File: library/caddy/2.7/README.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
+# Caddy 2.7 Image
+
+This directory contains the definition for the `unikraft.org/caddy:2.7` image running Nginx.
```
Comment (StefanJum) [inline library/caddy/2.7/README.md:3]:
```suggestion
This directory contains the definition for the `unikraft.org/caddy:2.7` image.
```

**Review by StefanJum — COMMENTED** [review]
One small comment, all good besides that.

**Review by StefanJum — COMMENTED** [review]
I get this error when running
```
en1: Interface is up
[    0.103447] ERR:  [libdevfs] Failed to create /dev: 20
[    0.106231] ERR:  [libukboot] Init function at 0x11bfa0 returned error -1
```

**Review by nderjung — APPROVED** [review]
All good!

**Pull request merged**